File: java/sbur-redis/src/main/java/com/thehecklers/sburredis/Aircraft.java
package com.thehecklers.sburredis;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;

import java.time.Instant;

@Data
@NoArgsConstructor
@AllArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
public class Aircraft {
    @Id
    private Long id;
    private String callsign, squawk, reg, flightno, route, type, category;
    private int altitude, heading, speed;
    @JsonProperty("vert_rate")
    private int vertRate;
    @JsonProperty("selected_altitude")
    private int selectedAltitude;
    private double lat, lon, barometer;
    @JsonProperty("polar_distance")
    private double polarDistance;
    @JsonProperty("polar_bearing")
    private double polarBearing;
    @JsonProperty("is_adsb")
    private boolean isADSB;
    @JsonProperty("is_on_ground")
    private boolean isOnGround;
    @JsonProperty("last_seen_time")
    private Instant lastSeenTime;
    @JsonProperty("pos_update_time")
    private Instant posUpdateTime;
    @JsonProperty("bds40_seen_time")
    private Instant bds40SeenTime;

    public String getLastSeenTime() {
        return lastSeenTime.toString();
    }

    public void setLastSeenTime(String lastSeenTime) {
        if (null != lastSeenTime) {
            this.lastSeenTime = Instant.parse(lastSeenTime);
        } else {
            this.lastSeenTime = Instant.ofEpochSecond(0);
        }
    }

    public String getPosUpdateTime() {
        return posUpdateTime.toString();
    }

    public void setPosUpdateTime(String posUpdateTime) {
        if (null != posUpdateTime) {
            this.posUpdateTime = Instant.parse(posUpdateTime);
        } else {
            this.posUpdateTime = Instant.ofEpochSecond(0);
        }
    }

    public String getBds40SeenTime() {
        return bds40SeenTime.toString();
    }

    public void setBds40SeenTime(String bds40SeenTime) {
        if (null != bds40SeenTime) {
            this.bds40SeenTime = Instant.parse(bds40SeenTime);
        } else {
            this.bds40SeenTime = Instant.ofEpochSecond(0);
        }
    }
}

File: chapter6end/kotlin/sbur-redis/src/main/kotlin/com/thehecklers/sburredis/Aircraft.kt
package com.thehecklers.sburredis

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import org.springframework.data.annotation.Id
import java.time.Instant

@JsonIgnoreProperties(ignoreUnknown = true)
data class Aircraft(
    @Id @JsonProperty("id") val id: Long? = 0L,
    @JsonProperty("callsign") val callsign: String? = "",
    @JsonProperty("squawk") val squawk: String? = "",
    @JsonProperty("reg") val reg: String? = "",
    @JsonProperty("flightno") val flightno: String? = "",
    @JsonProperty("route") val route: String? = "",
    @JsonProperty("type") val type: String? = "",
    @JsonProperty("category") val category: String? = "",
    @JsonProperty("altitude") val altitude: Int? = 0,
    @JsonProperty("heading") val heading: Int? = 0,
    @JsonProperty("speed") val speed: Int? = 0,
    @JsonProperty("vert_rate")
    val vertRate: Int? = 0,
    @JsonProperty("selected_altitude")
    val selectedAltitude: Int? = 0,
    @JsonProperty("lat") val lat: Double? = 0.0,
    @JsonProperty("lon") val lon: Double? = 0.0,
    @JsonProperty("barometer") val barometer: Double? = 0.0,
    @JsonProperty("polar_distance")
    val polarDistance: Double? = 0.0,
    @JsonProperty("polar_bearing")
    val polarBearing: Double? = 0.0,
    @JsonProperty("is_adsb")
    val isADSB: Boolean? = false,
    @JsonProperty("is_on_ground")
    val isOnGround: Boolean? = false,
    @JsonProperty("last_seen_time")
    val lastSeenTime: String? = Instant.ofEpochSecond(0).toString(),
    @JsonProperty("pos_update_time")
    val posUpdateTime: String? = Instant.ofEpochSecond(0).toString(),
    @JsonProperty("bds40_seen_time")
    val bds40SeenTime: String? = Instant.ofEpochSecond(0).toString()
)
